### ISSUE TITLE
Can't access windows after disconnecting from workstation (multiple monitors)

### ISSUE THREAD

**Issue by husamrahman** [material-shell/material-shell#647]
**Describe the bug**

I lose access to my workspaces that were being displayed on exrternal monitors when I unplug my laptop from my workstation. They still run in the background but there is no way to bring them back to the main screen. I have to close them out and re-open everything again in a new session.

**Expected behavior**
Everything thats open on different monitors should be availalbe on the primary monitor when disconnected from the other dispalys.

**To Reproduce**
- Occurs on:
  - [x] X.Org
  - [ ] Wayland
- GNOME Shell version: 3.36+
- Distro: Arch
- Steps to reproduce, if applicable:
1. Open windows on different monitors
2. Disconnect external monitors
3. Wont be able to bring windows back to main monitor

**Comment by HaukurPall** [comment]
I'm quite sure this is related to #571 and #576

**Comment by apachler** [comment]
I can confirm this issue on
- GNOME Shell version: 3.38
- Distro: Manjaro

**Comment by rizajur** [comment]
@husamrahman @apachler the issue is quite old, would someone of you be so kind and tell us if this issue exist in newer gnome versions ?

**Comment by apachler** [comment]
Yes, this issue still exists with Gnome Shell v40/42/43 and Wayland/Xorg.

Sometimes the whole system crashes or the Gnome session gets killed. That happens also without changing the monitor connections, just moving the windows between monitors/workspaces.

**Comment by PapyElGringo** [comment]
Did you tried with the latest changes on Main?
